dali/base/dadfs.cpp

```diff
@@ -1904,22 +1904,23 @@ class CDFScopeIterator: implements IDFScopeIterator, public CInterface
     {
         includeempty = _includeempty;
         dir = _dir;
-        StringBuffer baseq;
-        StringBuffer tmp;
-        if (base&&*base) {
+        StringBuffer lockPath;
+        if (!isEmptyString(base))
+        {
             CDfsLogicalFileName dlfn;
-            dlfn.set(base,".");
-            dlfn.makeScopeQuery(baseq,false);
+            dlfn.set(base, "dummyfilename"); // makeScopeQuery expects a lfn to a file, 'dummyfilename' will not be used
```

Comment:
Don't understand the not be used comment

Reply:
dlfn.set wants a base scope path (e.g. scope1::scope2) and a trailing logical filename.
But in this context 'dlfn' is only used to get the xpath for the scope, the tail logicalname is not used, hence dummy.